api/src/main/java/javax/enterprise/inject/spi/BeanManager.java

```diff
@@ -618,4 +620,18 @@
      */
     <T> InterceptionFactory<T> createInterceptionFactory(CreationalContext<T> ctx, Class<T> clazz);
 
+
+    /**
+     *
+     * Obtains an {@link Instance} giving access to beans instances
+     *
+     * Instances of dependent scoped beans obtained with this Instance should be explicitly release with {@link Instance#destroy(Object)}
+     *
+     * @return an {@link Instance} object to request beans instances
+     * @throws IllegalStateException if called during application initialization, before the {@link AfterDeploymentValidation}
+     *         event is fired.
+     * @since 2.0
+     */
+    Instance<Object> getInstance();
```

Comment:
IMHO, conveniency adds value from the developer experience standpoint. I'd rather prefer writing:

    manager.getInstance().select(type).get();

instead of:

    Bean<?> bean = manager.resolve(manager.getBeans(type, Any.Literal.INSTANCE));
    manager.getReference(bean, type, manager.createCreationalContext(bean));

or to have to create my util class in each extension that I develop, or to depends on DeltaSpike just for that purpose.

I feel a bit bad to be honest each time I present CDI extensions to people and have to find some convoluted way to retrieve contextual instances 😉.